code/modules/antagonists/traitor/components/demoraliser.dm

```diff
@@ -44,8 +44,13 @@
 	// If you're not conscious you're too busy or dead to look at propaganda
 	if (viewer.stat != CONSCIOUS)
 		return
+	if(viewer.is_blind())
+		return
 	if (!should_demoralise(viewer))
 		return
+	if(!viewer.can_read(host, TRUE, moods.text_based, FALSE))
```

Comment:
```suggestion
	if(moods.text_based && !viewer.can_read(host))
```

TRUE/FALSE for args is not good.  You also should be checking if it has the literacy check *outside* the `can_read` scope since you are just skipping the entire proc.  (and if you do need to use the proc, the `to_chat` messages shouldn't be silent)

Reply:
I'm not skipping the entire proc, I'm still using it to check light levels. Posters are not the only demoralizing thing. Not all demoralizing atom datums have text_based = true.